### PR TITLE
Replace the giant comment in routes.rb with a link to the guides

### PR DESCRIPTION
**Pull request by sgrif**
This comment not only serves no purpose, but in my experience is
actively detrimental to new developers getting started with Rails.
Expereinced developers just end up deleting this comment, and are
annoyed that they had to take this step. I also spend a lot of time
mentoring brand new developers, and a consistent theme I've seen is that
this comment just ends up intimidating them, and making them think it's
dangerous to edit this file.

One of my students just said this (due to the number of comments which
even new developers don't actually read, they just see it as a sign that
this thing is "dangerous").

> I don't edit any file that Rails generates for me, until my instructor
> says that it's OK to do so.

Realistically, this comment adds 0 value. We have very good
documentation, which we can just link to instead. If someone is truly
new enough to benefit from this info, they presumably just ran `gem
install rails`, and have an internet connection that they can use to
read the routing guide.

The choice of language here was very specific. I chose "the DSL
available" over "what is possible", because a consistent theme I've
noticed among my students is that they aren't aware that this is
actually a Ruby file, and can write any Ruby code here that they want.

This file is not the only offender, but is by far the biggest point of
pain that I've seen, and felt it was a good spot to open this
discussion.

### CODE REVIEW

File: railties/lib/rails/generators/rails/app/templates/config/routes.rb [hunk 1]
```diff
@@ -1,56 +1,3 @@
...
-  #   resources :posts, concerns: :toggleable
-  #   resources :photos, concerns: :toggleable
-
-  # Example resource route within a namespace:
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
```
Comment (edwardloveall) [inline railties/lib/rails/generators/rails/app/templates/config/routes.rb]: If a major goal of this is to not intimidate new users, I'd suggest something more friendly (at least with no acronyms) like:

>For details on how to use the routes file, see http://...

**Comment by edwardloveall** [comment]
As someone who recently switched over to the "less comments are better" camp I like this and pretty much agree with your points. Someone recently said "Comments are future lies" referring to upgrading versions of rails where some of these APIs go out of date.

**Comment by stevekinney** [comment]
As someone who teaches Rails for a living, I am a huge fan of this change (and hopefully future changes to the `Gemfile` and the rest). For new developers it's overwhelming and it also distracts from the official documentation—which is a much better resource for newer developers. For experiences developers, it's just another thing to remove when starting a new project.

I love, love, love that it was replaced with a link to the official docs.

File: railties/lib/rails/generators/rails/app/templates/config/routes.rb (hunk 1 above)
Comment (sgrif) [inline railties/lib/rails/generators/rails/app/templates/config/routes.rb]: I'm fine with that. Any wording which doesn't contradict the fact that this is a Ruby file (as mentioned in the commit message)

**Comment by mikedao** [comment]
I am in complete agreement with @stevekinney. I also teach Rails, and feel that people new to Rails upon seeing the commented out portions will feel quite overwhelmed. However a link that simply points to the documentation, and I suggest that we change it to @edwardloveall 's suggestion, avoiding acronyms, will help people not only reach familiarity, but perhaps get them into the habit of reading the very well written documentation. 

And on the other hand, it's also one less thing people that do have some familiarity with Rails have to remove when building an app.

**Comment by jasonnoble** [comment]
As an instructor, I agree that these comments are rarely useful.  A link to the guide is the way to go.

A possible alternative would be to introduce a command line option to generate the full docs?  Maybe `--minimal-comments` or `--full-comments`?

**Comment by ianks** [comment]
I ACK this as well, but would ACK it more if you encouraged a student to make this PR 😀

File: railties/lib/rails/generators/rails/app/templates/config/routes.rb (hunk 1 above)
Comment (edwardloveall) [inline railties/lib/rails/generators/rails/app/templates/config/routes.rb]: Sure. Could be:
>For details on how to use the routes.rb file, see http://...

which might make it slightly more clear, but maybe the best is to make it clear in the documentation itself.

**Comment by EmilyMB** [comment]
As a newer developer, I agree with the comments above and :heart: @edwardloveall's friendly phrasing. All the comments throughout all of Rails seem verbose and unnecessary... If I get stuck on something, I just Google it. Between Stack Overflow and the docs, things are pretty well covered.

**Comment by jonatack** [comment]
Agreed concerning "comments throughout all of Rails" for generated files, but not in the codebase itself. For what its worth, I'd say the first/best/most reliable places to look for information are the Rails source code & comments (`bundle open rails` on the command line), api.rubyonrails.org, guides.rubyonrails.org, and for Ruby, http://ruby-doc.org.

**Comment by loriculberson** [comment]
Great idea! A link to the guides would provide a cleaner interface. I like Jason Noble's suggestion to offer the option of minimal or full comments upon installation. Though I don't know if anyone would choose to add them.

**Comment by marlabrizel** [comment]
:+1: to this change. As a new Rails dev, I actually fear inadvertently deleting configuration code when I'm deleting large sections of comments. While this is likely less of an issue with the routes file, I can envision a scenario elsewhere in which I delete commented out code, accidentally nuke something essential to my config, and then spend lots of time debugging due to my still-limited familiarity with the framework.

**Comment by rafaelfranca** [comment]
Yeah, I agree with the change, but I'd like to hear @fxn and @dhh comments.

**Comment by dhh** [comment]
I'm fine replacing this with a link to a guide, but let's not kid ourselves thinking that the guide is somehow going to be magically more up to date than a code comment. Also, I don't agree that comments in general are bad in generated files because you can google things and maybe find something on Stack Overflow. But in this case, this isn't a comment, but literally a full guide. Don't need that.

**Comment by sgrif** [comment]
Sounds like we're all on the same page. I'm going to move forward with this.

**Comment by fxn** [comment]
Agree with David's arguments.

The link points to the latest stable guide, it would be better to point to the versioned one.

**Pull request merged**